ggml/src/CMakeLists.txt

```diff
@@ -305,6 +305,13 @@ else ()
     ggml_add_cpu_backend_variant_impl("")
 endif()
 
+if (GGML_OPENCL)
+    set(GGML_OPENCL2 ON)
+    add_compile_definitions(GGML_USE_OPENCL)
+else ()
+    set(GGML_OPENCL2 OFF)
+endif ()
```

Comment:
What's the reason for calling this backend "opencl2"? I think this can create some confusion since there isn't a "opencl1" backend. It would be preferable to rename it to just `ggml-opencl` and remove this. This is not correct as it is, since `GGML_USE_OPENCL` should not be defined if `GGML_BACKEND_DL` enabled.

Reply:
Technically the backend is targeting OpenCL 2.x but yeah, we should just call it OpenCL. 
I thought I tested `GGML_BACKEND_DL`.   
Will rename & fix shortly.

Reply:
I think the problem is that there is a discrepancy between the backend directory name and the macro (`GGML_OPENCL_xxx`) and also the backend shared library name. This portion of code breaks `ggml_add_backend_library`, which automatically defines `GGML_USE_xxx` based on the directory name.

We will rename the backend directory to `ggml-opencl` so this portion can be removed; we will keep `opencl2` in the function names in the backend implementation for now.

There used to be an OpenCL backend based on CLBlast, it was named `ggml-opencl` (the macro was `GGML_USE_CLBLAST`, so we are able to use `GGML_USE_OPENCL`). Therefore, we named ours `ggml-opencl2`; coincidentally, OpenCL 2.0 or above is required.